packages/engine-test-utils/src/__tests__/engine-server/EngineEvents.spec.ts

```diff
@@ -0,0 +1,404 @@
+import { NoteChangeEntry, NoteChangeUpdateEntry } from "@dendronhq/common-all";
+import { NoteTestUtilsV4 } from "@dendronhq/common-test-utils";
+import { DendronEngineClient } from "@dendronhq/engine-server";
+import { createEngineFromServer, ENGINE_HOOKS, runEngineTestV5 } from "../..";
+
+/**
+ * Tests that the EngineEvents interface of the client-side DendronEngineClient signals properly
+ */
+describe("GIVEN a DendronEngineClient running on client-side", () => {
+  describe("WHEN writing a new note", () => {
+    test("THEN create event fired with the correct props AND update note event fired for its parent", async (done) => {
+      await runEngineTestV5(
+        async ({ engine, vaults, wsRoot }) => {
+          // TODO: Get rid of this casting once the proper way to expose EngineEvents of DendronEngineClient is implemented
+          const engineClient = engine as DendronEngineClient;
+
+          const newNote = await NoteTestUtilsV4.createNote({
+            fname: "alpha",
+            body: `[[beta]]`,
+            vault: vaults[0],
+            wsRoot,
+          });
+
+          engineClient.onEngineNoteStateChanged(
+            (noteChangeEntries: NoteChangeEntry[]) => {
+              try {
+                expect(noteChangeEntries.length).toEqual(2);
+
+                noteChangeEntries.forEach((entry) => {
```

Comment:
this is hard to read. can we add a helper function, especially since `NoteChangeUpdateEntry` will be used quite a bit?

```
extractEntries(status) => NoteChangeUpdateEntry[]
```

we can add it to `engine-test-utils/src/engine.ts`

Reply:
apply feedback to subsequent tests